stablehlo/reference/Element.cpp

```diff
@@ -175,63 +175,57 @@ bool areApproximatelyEqual(APFloat f, APFloat g) {
 
 }  // namespace
 
-Element::Element(Type type, APInt value) : type_(type), value_(value) {}
-
-Element::Element(Type type, int64_t value) {
+Element::Element(Type type, APInt value) {
   if (!isSupportedIntegerType(type))
     report_fatal_error(invalidArgument("Unsupported element type: %s",
                                        debugString(type).c_str()));
+  if (type.getIntOrFloatBitWidth() != value.getBitWidth())
+    report_fatal_error(
+        invalidArgument("Bitwidth mismatch. Type: %s, Value: %s",
+                        debugString(type.getIntOrFloatBitWidth()).c_str(),
+                        debugString(value.getBitWidth()).c_str()));
   type_ = type;
-  value_ = APInt(type.getIntOrFloatBitWidth(), value,
-                 /*isSigned=*/isSupportedSignedIntegerType(type));
+  value_ = value;
 }
 
-Element::Element(Type type, bool value) : type_(type), value_(value) {}
-
-Element::Element(Type type, APFloat value) : type_(type), value_(value) {}
-
-Element::Element(Type type, double value) {
-  if (isSupportedFloatType(type)) {
-    APFloat floatVal(value);
-    bool roundingErr;
-    floatVal.convert(type.cast<FloatType>().getFloatSemantics(),
-                     APFloat::rmNearestTiesToEven, &roundingErr);
-    type_ = type;
-    value_ = floatVal;
-  } else if (isSupportedComplexType(type)) {
-    APFloat real(value);
-    APFloat imag(0.0);
-    auto floatTy = type.cast<ComplexType>().getElementType().cast<FloatType>();
-    bool roundingErr;
-    real.convert(floatTy.getFloatSemantics(), APFloat::rmNearestTiesToEven,
-                 &roundingErr);
-    imag.convert(floatTy.getFloatSemantics(), APFloat::rmNearestTiesToEven,
-                 &roundingErr);
-    type_ = type;
-    value_ = std::make_pair(real, imag);
-  } else {
+Element::Element(Type type, bool value) {
+  if (!isSupportedBooleanType(type))
     report_fatal_error(invalidArgument("Unsupported element type: %s",
                                        debugString(type).c_str()));
-  }
+  if (type.getIntOrFloatBitWidth() != 1)
```

Comment:
`value` here is a primitive `bool`, and sizeof(bool) returns 1 (1 byte, so 8 bits not 1 bit as StableHLO represents, and checking for `type.getIntOrFloatBitWidth() != sizeof(bool)` is not correct imo). Providing this under namespace is the closest I could think of to avoid this magic number but prevent overhead:
```
constexpr uint64_t boolBitWidth = 1;
```

Reply:
I see. Nevermind, I thought the value is `APInt`.